src/emulator/storage/apis/firebase.ts

```diff
@@ -356,8 +357,9 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
         }
         throw err;
       }
-      metadata.addDownloadToken();
-      return res.status(200).json(new OutgoingFirebaseMetadata(metadata));
+
+      storedMetadata.addDownloadToken();
```

Comment:
I'm surprised this isn't failing an integration test somewhere since `addDownloadToken` calls `update({})` internally. Can you make sure we cover this in a test case that this doesn't trigger a metadata update cloud function as well

Reply:
Synced offline. It will take some work to get tests working that also account for the number of times different cloud functions are called specifically from Storage. I've added a TODO to capture the fact we need to test for trigger counts in these tests as well.

In the meanwhile I've amended the PR to pass in the flag in the correct spots.